### PR TITLE
fix: @effection/core to ship dist-esm and dist-cjs

### PR DESCRIPTION
**Pull request by jbolda**
## Motivation

This supersedes #464. Copying the `package.json` is causing deeper issues.

## Approach

We instead use `dist-esm` and `dist-cjs` and this keeps the same folder depth to the root as with `src`.

### CODE REVIEW

**Review by cowboyd — COMMENTED** [review]
We need to be consistent on how we handle this for all packages. In other words, we should use `dist-esm` and `dist-cjs` across the board.

**Comment by jbolda** [comment]
> We need to be consistent on how we handle this for all packages. In other words, we should use `dist-esm` and `dist-cjs` across the board.

Sure, I can change everything over.

Looking into why these tests are failing as well.

**Comment by dagda1** [comment]
> > We need to be consistent on how we handle this for all packages. In other words, we should use `dist-esm` and `dist-cjs` across the board.

I still think rollup is the best answer for building typescript packages with its tree-shaking support but I understand the complexity rises with these monsters.

I'll try not to mention this again :).

**Comment by cowboyd** [comment]
> I still think rollup is the best answer for building typescript packages with its tree-shaking support but I understand the complexity rises with these monsters.

I like rollup too, but  if we take a slim approach and just ship unbundled javascript, then you don't end up in a situation where you have have rollups within rollups within webpacks....

File: packages/atom/package.json [hunk 1]
```diff
@@ -2,21 +2,22 @@
...
   },
-  "typeDocEntry": "src/index.ts",
   "author": "Frontside Engineering <engineering@frontside.io>",
   "license": "MIT",
   "files": [
     "README.md",
     "CHANGELOG.md",
-    "dist/**/*",
+    "dist-esm/**/*",
+    "dist-cjs/**/*",
```
Comment (cowboyd) [inline packages/atom/package.json]: We could make this a little more DRY
```suggestion
    "dist-*/**/*",
```

Comment (jbolda) [reply packages/atom/package.json]: oh that is true :+1:

**Comment by frontsidejack** [comment]
The following preview packages have been published by Covector:
- `@effection/channel@2.0.0-fix-core-ship-esm-cjs-dist-r2.1628053316093`
- `@effection/atom@2.0.0-fix-core-ship-esm-cjs-dist-r2.1628053316093`
- `@effection/core@2.0.0-fix-core-ship-esm-cjs-dist-r2.1628053316093`
- `effection@2.0.0-fix-core-ship-esm-cjs-dist-r2.1628053316093`
- `@effection/events@2.0.0-fix-core-ship-esm-cjs-dist-r2.1628053316093`
- `@effection/fetch@2.0.0-fix-core-ship-esm-cjs-dist-r2.1628053316093`
- `@effection/inspect@2.0.0-fix-core-ship-esm-cjs-dist-r2.1628053316093`
- `@effection/inspect-server@2.0.0-fix-core-ship-esm-cjs-dist-r2.1628053316093`
- `@effection/inspect-utils@2.0.0-fix-core-ship-esm-cjs-dist-r2.1628053316093`
- `@effection/main@2.0.0-fix-core-ship-esm-cjs-dist-r2.1628053316093`
- `@effection/node@2.0.0-fix-core-ship-esm-cjs-dist-r2.1628053316093`
- `@effection/mocha@2.0.0-fix-core-ship-esm-cjs-dist-r2.1628053316093`
- `@effection/process@2.0.0-fix-core-ship-esm-cjs-dist-r2.1628053316093`
- `@effection/react@2.0.0-fix-core-ship-esm-cjs-dist-r2.1628053316093`
- `@effection/subscription@2.0.0-fix-core-ship-esm-cjs-dist-r2.1628053316093`
- `@effection/websocket-client@2.0.0-fix-core-ship-esm-cjs-dist-r2.1628053316093`
- `@effection/websocket-server@2.0.0-fix-core-ship-esm-cjs-dist-r2.1628053316093`
- `@effection/inspect-ui@2.0.0-fix-core-ship-esm-cjs-dist-r2.1628053316093`

<p align="right">6923a0f</p>

**Review by cowboyd — APPROVED** [review]
🥳

**Pull request merged**